details movies implemented

briansetzesglobant opened the pull request:
Perform the part of the implementation of the movie project, that clicking on a movie brings up the list of details

Video of the APP running
https://user-images.githubusercontent.com/84338449/123455855-91198680-d5b8-11eb-87d3-a45520fb0198.mp4

briansetzesglobant committed:
details movies implemented

FranciscoGaraventa commented on lib/src/models/movie_result.dart:
```diff
@@ -1,67 +1,66 @@
+    this.voteCount,
+    this.id,
+    this.video,
+    this.voteAverage = 0.0,
```
Add this initial value to constant file like voteAverage

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/models/movie_result.dart:
```diff
@@ -1,67 +1,66 @@
+    this.id,
+    this.video,
+    this.voteAverage = 0.0,
+    this.title = "",
```
This can be added to String file, like emptyString

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/ui/home_detail_page.dart:
```diff
@@ -0,0 +1,55 @@
+    return Scaffold(
+      backgroundColor: Colors.blueGrey,
+      bottomNavigationBar: Padding(
+        padding: const EdgeInsets.all(Constants.paddingHomeDetails),
```
You missed a trailing comma here

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/ui/home_page.dart:
```diff
@@ -108,25 +109,40 @@ class HomePageState extends State<HomePage> {
+                  movieReleaseDate:
+                      movieData.results[index].releaseDate!.isNotEmpty
+                          ? movieData.results[index].releaseDate!
+                          : "Without date",
```
This can be a String constant

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/widgets/movie_detail_button.dart:
```diff
@@ -0,0 +1,46 @@
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(Constants.paddingButtonChildren),
```
You missed a trailing comma here

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/widgets/movie_detail_button.dart:
```diff
@@ -0,0 +1,46 @@
+              ),
+            ),
+            Text(
+              "BACK",
```
This can be added to String file

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/widgets/movie_detail_image.dart:
```diff
@@ -0,0 +1,26 @@
+        Constants.paddingImage,
+      ),
+      child: FadeInImage.assetNetwork(
+        placeholder: posterPath,
+        image: posterPath,
```
Why you use placeholder and image with the same path?

FranciscoGaraventa replied:
You can add a specific route to placeholder, with another image that don't have relation with posterPath that you want to show

briansetzesglobant replied:
Because the default image already came in posterPath

FranciscoGaraventa commented on lib/src/widgets/movie_detail_information.dart:
```diff
@@ -0,0 +1,34 @@
+      child: Column(
+        children: [
+          Text(
+            "Date: $date",
```
You can add this to String file like movieDetailDateText = "Date: " and movieDetailRatingText = "Rating: "

briansetzesglobant replied:
done

FranciscoGaraventa commented:
As a suggestion you could adapt the image shown on the main screen so that it does not appear cut off

briansetzesglobant committed:
changes fixed
            Some variables have been renamed to a more descriptive name
Elimination of a comma

FranciscoGaraventa commented on lib/src/models/movie.dart:
```diff
@@ -1,28 +1,32 @@
+    var parsedJsonList = parsedJson['results'] as List;
 
-  int get total_results => _total_results;
+    List<MovieResult> movieTemporary = parsedJsonList
```
Instead of movieTemporary do you think that is more descriptive to change it for movieResultList? Or something like that.

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/models/movie_result.dart:
```diff
@@ -1,67 +1,67 @@
+  });
+
+  factory MovieResult.fromJson(Map<String, dynamic> parsedJson) {
+    var parsedJsonList = parsedJson['genre_ids'] as List;
```
Similar to last comment, this can be genresJsonList or something like that

briansetzesglobant replied:
done

FranciscoGaraventa commented on lib/src/resources/movie_strings.dart:
```diff
@@ -11,4 +11,10 @@ abstract class MovieStrings {
       "https://kubalubra.is/wp-content/uploads/2017/11/default-thumbnail.jpg";
+
+  static const String emptyString = "";
+  static const String withoutDate = "Without date";
```
This string can be movieDefaultDate

briansetzesglobant replied:
done

MarcosCGlobant reviewed (approved):
LGTM

FranciscoGaraventa commented on lib/src/models/movie.dart:
```diff
@@ -1,28 +1,32 @@
 
-  int get total_pages => _total_pages;
+  factory Movie.fromJson(
+    Map<String, dynamic> parsedJson,
```
This comma after parsedJson is not necessary

briansetzesglobant replied:
done

FranciscoGaraventa reviewed (approved):
LGTM
